launcher/CMakeLists.txt

```diff
@@ -1093,6 +1105,52 @@ install(TARGETS ${Launcher_Name}
     FRAMEWORK DESTINATION ${FRAMEWORK_DEST_DIR} COMPONENT Runtime
 )
 
+if(WIN32)
+    add_library(filelink_logic STATIC ${LINKEXE_SOURCES})
+    target_include_directories(filelink_logic PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+    target_link_libraries(filelink_logic
+        systeminfo
+        BuildConfig
+        Qt${QT_VERSION_MAJOR}::Widgets
+        ghcFilesystem::ghc_filesystem
+    )
+    target_link_libraries(filelink_logic
+        Qt${QT_VERSION_MAJOR}::Core
+        Qt${QT_VERSION_MAJOR}::Xml
+        Qt${QT_VERSION_MAJOR}::Network
+        Qt${QT_VERSION_MAJOR}::Concurrent
+        ${Launcher_QT_LIBS}
+    )
+
+    add_executable("${Launcher_Name}_filelink" WIN32 filelink/main.cpp)
+
+    target_sources("${Launcher_Name}_filelink" PRIVATE filelink/filelink.exe.manifest)
+
+    target_link_libraries("${Launcher_Name}_filelink" filelink_logic)
+
+    if(DEFINED Launcher_APP_BINARY_NAME)
+        set_target_properties("${Launcher_Name}_filelink" PROPERTIES OUTPUT_NAME "${Launcher_APP_BINARY_NAME}_filelink")
+    endif()
+    if(DEFINED Launcher_BINARY_RPATH)
+        SET_TARGET_PROPERTIES("${Launcher_Name}_filelink" PROPERTIES INSTALL_RPATH "${Launcher_BINARY_RPATH}")
+    endif()
+
+    # may be unnessacery with manifest
+    if(CMAKE_GENERATOR MATCHES "Visual Studio")
+        SET_TARGET_PROPERTIES("${Launcher_Name}_filelink" PROPERTIES LINK_FLAGS "/level='requireAdministrator' /uiAccess='false' /SUBSYSTEM:CONSOLE")
+    # else()  # link arg /MANIFESTUAC only works with MSVC
```

Comment:
how does it work on mingw then?

Reply:
Technically I'm not even sure using this linker arg adds anything meaningful. asking the exe to request admin privileges on run doesn't work in our use context. This support *should* be added by the `.manifest` anyway so the use of the linker arg is redundant.

Reply:
If it's already on the manifest can't we just yeet that?

Reply:
Yeah, I'll remove it in the next commits I push.